src/common/Platform.h

```diff
@@ -63,15 +63,73 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #define __x86_64__ 1
 #endif
 
-// SSE support
-#if defined(__x86_64__) || defined(__SSE__) || _M_IX86_FP >= 1
-#include <xmmintrin.h>
-#if defined(__x86_64__) || defined(__SSE2__) || _M_IX86_FP >= 2
-#include <emmintrin.h>
-#define idx86_sse 2
-#else
-#define idx86_sse 1
+/* Detect MSVC providing SSE and SSE2.
+
+MSVC doesn't set __SSE*__, and only sets _M_IX86_FP on i686.
+We should look for _M_AMD64 or _M_X64 to know if SSE and SSE2
+are enabled when building code for amd64. Beware that _M_AMD64
+and _M_X64 are also enabled when building for ARM64EC:
+
+> - _M_AMD64 Defined as the integer literal value 100 for compilations
+>   that target x64 processors or ARM64EC. Otherwise, undefined.
+> - _M_X64 Defined as the integer literal value 100 for compilations
+>   that target x64 processors or ARM64EC. Otherwise, undefined.
+> - _M_ARM64EC Defined as 1 for compilations that target ARM64EC.
+>   Otherwise, undefined.
+> -- https://learn.microsoft.com/en-us/cpp/preprocessor/predefined-macros?view=msvc-170
+
+It is unclear if xmmintrin.h is available on ARM64EC. */
+#if defined(_MSC_VER)
+	#if defined(_M_IX86_FP)
+		#if _M_IX86_FP >= 2
+			#define MSVC_SSE
```

Comment:
It seems better to define `DAEMON_USE_ARCH_INTRINSICS_i686_sse` etc. directly here, rather than polluting the namepace with another layer of `MSVC_SSE` macros.

Reply:
Done.

I could also set `__SSE__` and `__SSE2__` instead of custom `MSVC_SSE` and `MSVC_SSE2` so the rest of the code would be generic.

Reply:
> I could also set `__SSE__` and `__SSE2__` instead of custom `MSVC_SSE` and `MSVC_SSE2` so the rest of the code would be generic.

In the same way we define `__x86_64__` in all cases, right above.

Reply:
I added a commit with that other idea, tell me what you think about it.

This _may_ not be wanted if it interfers badly with third-party libraries.

Reply:
It's not good to define those macros which are supposed to be defined by the compiler. Technically it's undefined behavior to define anything beginning with double underscore or underscore, capital letter.

Reply:
OK,  I went back to the `ifdef` tree.